capstone/src/main/webapp/js/calendar.js

```diff
@@ -0,0 +1,66 @@
+/* eslint no-unused-vars: ["error", {"varsIgnorePattern": "drawCharts"}] */
+
+/**
+ * Callback function to generate charts.
+ */
+function drawCharts() {
+  const frequencyData = new google.visualization.DataTable();
+  fetch('/viz/deadlines').
+      then((response) => response.json()).
+      then((deadlineInfo) =>
+        populateDataTables(frequencyData, deadlineInfo)).
+      then(() => drawFrequencyChart(frequencyData));
+}
+
+/**
+ * Generate frequency calendar chart of deadlines.
+ * @param {google.visualization.DataTable} frequencyData the DataTable
+ *     corresponding to the frequency of deadlines
+ */
+function drawFrequencyChart(frequencyData) {
+  const options = {
+    'title': 'Expected Application Heat Map',
+  };
+  const chart = new google.visualization.Calendar(
+      document.getElementById('calendar-freq-map'));
+  chart.draw(frequencyData, options);
+}
+
+/**
+ * Helper function to transfer elements from JSON object with deadline
+ * information to a DataTable.
+ *
+ * @param {google.visualization.DataTable} frequencyData the DataTable with the
+ *     data the frequency chart will use. Will consist of sample deadlines for
+ *     each of the user's selected colleges.
+ *
+ * @param {JSON} deadlineInfo the JSON object with the deadline information that
+ *     will be added to the previously mentioned DataTables.
+ *
+ */
+function populateDataTables(frequencyData, deadlineInfo) {
+  frequencyData.addColumn('date', 'Date');
+  frequencyData.addColumn('number', 'Application Opens/Closes');
+  deadlineInfo.forEach((college) => {
+    frequencyData.addRow(
+        [
+          new Date(
+              college['openingDate']['year'],
+              college['openingDate']['month'] - 1,
```

Comment:
It's not immediately clear to me why you're adding a `- 1` to the month. Is it because one starts from zero and the other starts at index 1? Add a comment to clarify.

Reply:
Ah yes Javascript months start from 0, but Java months start from 1. I'll add a comment to make that clearer.